packages/tc-api-rest-handlers/lib/relationships/input.js

```diff
@@ -125,9 +195,34 @@ const containsRelationshipData = (type, payload = {}) => {
 		.some(([propName]) => propName in payload || `!${propName}` in payload);
 };
 
+// relationship value could be just a String(code), an Object or an Array of strings(codes) / objects(code and properties) / mixed
+// this function is to normalise them into an Array of objects
+const normaliseRelationshipProps = (type, body) => {
+	const relationships = getWriteAndDeleteRelationships({ type, body });
```

Comment:
having read the code and seen what this is doing, I think... hmmm... it is definitely a case of "Naming things is hard!"... but I will have a think about what might be clearer names.

Reply:
Have left some suggestions. 

But another - maybe simpler - way of dealing with this could be to not bother with filtering out relationshipProperties which just use `null`. So you can iterate through ALL relationship properties that are in the body, but just add a rule in this function that if the value is `null`, then return it without modification?